### PR TITLE
Security Login.

### PR DESCRIPTION
**Pull request by Danil98Borisov**
Security Login.

### CODE REVIEW

File: freedom-of-travel-api/src/main/java/nc/project/service/AuthService.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.http.ResponseEntity;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Map;
+
+public interface AuthService {
+
+    Map<String, Object> authenticateUser(LoginRequest loginRequest);
+
+    ResponseEntity<Map<String, Object>> registerUser(SignupRequest signUpRequest, String siteURL) throws UnsupportedEncodingException;
```
Comment (AndrewSmrnv) [inline freedom-of-travel-api/src/main/java/nc/project/service/AuthService.java:14]: Please remove the code which doesn't relate to the Login functionality

File: freedom-of-travel-impl/src/main/java/nc/project/service/AuthServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
+package nc.project.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import nc.project.auth.request.LoginRequest;
+import nc.project.auth.request.SignupRequest;
```
Comment (AndrewSmrnv) [inline freedom-of-travel-impl/src/main/java/nc/project/service/AuthServiceImpl.java:6]: Please remove the code which doesn't relate to the Login functionality

File: freedom-of-travel-impl/src/main/java/nc/project/service/AuthServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            });
+        }
+
+        user.setRoles(roles);
+        userRepository.save(user);
+
+        outRegisterUser.remove("message");
+        outRegisterUser.put("message", "User registered successfully!");
+        return new ResponseEntity<>(outRegisterUser, HttpStatus.OK);
+    }
```
Comment (AndrewSmrnv) [inline freedom-of-travel-impl/src/main/java/nc/project/service/AuthServiceImpl.java:120]: Please remove the code which doesn't relate to the Login functionality

File: freedom-of-travel-web/src/main/java/nc/project/controller/AuthController.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                                 userDetails.getUsername(),
+                                 userDetails.getEmail(),
+                                 roles));
+    }
+
+    @PostMapping("/signup")
+    public ResponseEntity<?> registerUser(@Valid @RequestBody SignupRequest signUpRequest,
+                                          HttpServletRequest request) throws UnsupportedEncodingException {
+        return authService.registerUser(signUpRequest, getSiteURL(request));
+    }
```
Comment (AndrewSmrnv) [inline freedom-of-travel-web/src/main/java/nc/project/controller/AuthController.java:46]: Please remove the code which doesn't relate to the Login functionality

File: freedom-of-travel-web/src/main/java/nc/project/controller/AuthController.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package nc.project.controller;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import nc.project.auth.request.LoginRequest;
+import nc.project.auth.request.SignupRequest;
```
Comment (AndrewSmrnv) [inline freedom-of-travel-web/src/main/java/nc/project/controller/AuthController.java:6]: Please remove the code which doesn't relate to the Login functionality

File: freedom-of-travel-api/src/main/java/nc/project/service/AuthService.java (hunk 1 above)
Comment (Danil98Borisov) [reply freedom-of-travel-api/src/main/java/nc/project/service/AuthService.java]: done

File: freedom-of-travel-impl/src/main/java/nc/project/service/AuthServiceImpl.java (hunk 2 above)
Comment (Danil98Borisov) [reply freedom-of-travel-impl/src/main/java/nc/project/service/AuthServiceImpl.java]: done

File: freedom-of-travel-web/src/main/java/nc/project/controller/AuthController.java (hunk 5 above)
Comment (Danil98Borisov) [reply freedom-of-travel-web/src/main/java/nc/project/controller/AuthController.java]: done

File: freedom-of-travel-web/src/main/java/nc/project/controller/AuthController.java (hunk 4 above)
Comment (Danil98Borisov) [reply freedom-of-travel-web/src/main/java/nc/project/controller/AuthController.java]: done

**Pull request merged**